### PR TITLE
fix: correctly hide project set's projected_row_id column

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
fix https://github.com/risingwavelabs/risingwave/issues/5284

### CODE REVIEW

File: src/frontend/src/planner/query.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl Planner {
         }
         let mut out_fields = FixedBitSet::with_capacity(plan.schema().len());
         out_fields.insert_range(..plan.schema().len() - extra_order_exprs_len);
-        if plan.as_logical_project_set().is_some() {
+        if let Some(field) = plan.schema().fields.get(0) && field.name == "projected_row_id"  {
```
Comment (TennyZhuang) [inline src/frontend/src/planner/query.rs:57]: Do we always assume that the `project_set` is the top level operator?

Comment (xxchan) [reply src/frontend/src/planner/query.rs:57]: We did, and this PR fixed that. See the planner test above, where `project_set` is not at the top level, but output column contains `projected_row_id`.

**Comment by xxchan** [comment]
ping @xiangjinwu for a review

**Review by xiangjinwu — APPROVED** [review]
lgtm

File: src/frontend/src/planner/query.rs (hunk 1 above)
Comment (xiangjinwu) [reply src/frontend/src/planner/query.rs:57]: So we are loosing the assumption to this: `projected_row_id` is always the 0-th column / there are no PlanNode after ProjectSet that can reorder the schema.

File: src/frontend/planner_test/tests/testdata/project_set.yaml [hunk 2]
```diff
@@ -81,3 +81,66 @@
       └─BatchProjectSet { select_list: [Generate($3, 100:Int32, 1:Int32)] }
         └─BatchProjectSet { select_list: [$0, $1, Unnest($0)] }
           └─BatchScan { table: t, columns: [t.x, t._row_id], distribution: UpstreamHashShard(t._row_id) }
+- name: issue-5284
+  sql: |
+    create table t(x int[]);
+    select unnest(x) as unnest from t order by unnest limit 1;
+  batch_plan: |
+    BatchProject { exprs: [Unnest($0)] }
+    └─BatchTopN { order: "[projected_row_id ASC]", limit: 1, offset: 0 }
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/project_set.yaml:90]: https://github.com/risingwavelabs/risingwave/issues/17448 😅

**Pull request merged**